src/ZPublisher/HTTPRequest.py

```diff
@@ -1644,18 +1644,19 @@ class TemporaryFileWrapper(_TemporaryFileWrapper):
     and doesn't complain about such a move either.
     """
 
-    unlink = staticmethod(unlink)
-    isfile = staticmethod(isfile)
+    if PY2:
+        unlink = staticmethod(unlink)
+        isfile = staticmethod(isfile)
 
-    def close(self):
-        if not self.close_called:
-            self.close_called = True
-            self.file.close()
+        def close(self):
+            if not self.close_called:
+                self.close_called = True
+                self.file.close()
 
-    def __del__(self):
-        self.close()
-        if self.isfile(self.name):
-            self.unlink(self.name)
+        def __del__(self):
+            self.close()
+            if self.isfile(self.name):
+                self.unlink(self.name)
```

Comment:
I've simply indented the entire class body into the PY2 block. The `close` method looks for `self.close_called`, which raises an AttributeError in Python 3. And the fix in Python 3 was about avoiding a `__del__` in the first place, and rather using a new `_TemporaryFileCloser`. So I think both `close/__del__` should be in the block. The unlink/isfile methods are only used by the `__del__`, so we don't need them either in Py3.